etl/steps/data/garden/war/2023-08-15/mie.py

```diff
@@ -0,0 +1,330 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from typing import cast
+
+import numpy as np
+import owid.catalog.processing as pr
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+# Logger
+log = get_logger()
+# Mapping from hostility level code to name
+HOSTILITY_LEVEL_MAP = {
+    1: "No militarized action",
+    2: "Threat to use force",
+    3: "Display of force",
+    4: "Use of force",
+    5: "War",
+}
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = cast(Dataset, paths.load_dependency("mie"))
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["mie"].reset_index()
+
+    #
+    # Process data.
+    #
+    log.info("war.mie: sanity checks")
+    _sanity_checks(tb)
+
+    log.info("war.mie: keep relevant columns")
+    COLUMNS_RELEVANT = [
+        "micnum",
+        "eventnum",
+        "ccode1",
+        "ccode2",
+        "styear",
+        "endyear",
+        "hostlev",
+        "fatalmin1",
+        "fatalmax1",
+        "fatalmin2",
+        "fatalmax2",
+    ]
+    tb = tb[COLUMNS_RELEVANT]
+
+    log.info("war.mie: rename columns")
+    tb = reshape_table(tb)
+
+    # Checks
+    assert not tb["fatalmin"].isna().any(), "Nulls found in fatalmin!"
+    assert not tb["fatalmax"].isna().any(), "Nulls found in fatalmax!"
+
+    log.info("war.mie: add regions")
+    tb = add_regions(tb)
+
+    log.info("war.mie: rename column")
+    tb = tb.rename(columns={"hostlev": "hostility_level"})
+
+    log.info("war.mie: aggregate entries")
+    tb = tb.groupby(["micnum", "region", "hostility_level", "styear", "endyear"], as_index=False).agg(
+        {"fatalmin": "sum", "fatalmax": "sum"}
+    )
+
+    log.info("war.mie: expand observations")
+    tb = expand_observations(tb)
+
+    # estimate metrics
+    log.info("war.mie: estimate metrics")
+    tb = estimate_metrics(tb)
+
+    # Rename hotility levels
+    log.info("war.mie: rename hostility_level")
+    tb["hostility_level"] = tb["hostility_level"].map(HOSTILITY_LEVEL_MAP | {"all": "all"})
+    assert tb["hostility_level"].isna().sum() == 0, "Unmapped regions!"
+    # Add suffix with source name
+    msk = tb["region"] != "World"
+    tb.loc[msk, "region"] = tb.loc[msk, "region"] + " (COW)"
```

Comment:
Yeah, that's a good one.

I am leaving the prefixes in Garden, so that the dataset makes sense on its own. But I remove it in Grapher, so that it looks better (less noisy) on explorers.